src/components/Login/PhoneForm/formfields.tsx

```diff
@@ -0,0 +1,120 @@
+import { useContext, useEffect, useState } from 'react'
+import { FirebaseError } from 'firebase/app'
+import {
+  ConfirmationResult,
+  getAuth,
+  RecaptchaVerifier,
+  signInWithPhoneNumber
+} from 'firebase/auth'
+import { useSetAtom } from 'jotai'
+
+import { panelAtom } from '@/atoms/login'
+import validationSchema from '@/components/Login/PhoneForm/validation'
+import Button from '@/components/UI/button'
+import { FormContext } from '@/components/UI/FormComponents/Form/context'
+import PhoneField from '@/components/UI/FormComponents/PhoneField'
+import TextField from '@/components/UI/FormComponents/TextField'
+
+export interface FormValues {
+  phoneNumber: string
+  otpCode?: string
+}
+
+const FormFields = () => {
+  const auth = getAuth()
+  const { trigger, setError, getValues } = useContext(FormContext)
+  const setPanel = useSetAtom(panelAtom)
+  const [otpShown, setOtpShown] = useState(false)
+  const [recaptcha, setRecaptcha] = useState<RecaptchaVerifier>()
+  const [result, setResult] = useState<ConfirmationResult>()
+
+  useEffect(() => {
+    setRecaptcha(
+      new RecaptchaVerifier(
+        'recaptcha-container',
+        {
+          size: 'invisible'
+        },
+        auth
+      )
+    )
+  }, [auth])
+
+  const verifyPhone = async () => {
+    const valid = await trigger?.('phoneNumber', { shouldFocus: true })
+    const phoneNumber: string = getValues?.('phoneNumber')
+
+    if (!valid)
+      return setError?.('phoneNumber', { message: 'Invalid phone number' })
+
+    if (recaptcha) {
+      signInWithPhoneNumber(auth, phoneNumber, recaptcha)
+        .then((confirmationResult) => {
```

Comment:
Purely a preference thing, but you can use `await` here if you want, the function is already marked `async`.